src/action_cc.cc

```diff
@@ -375,13 +375,27 @@ NM_ACTION_(power) {
 
 NM_ACTION_(cmd_spawn) {
     #define NM_ERR_RET nullptr
+
     char *tmp = strdup(arg);
+
     char *cmd = tmp;
+    char *tmp1 = strtrim(strsep(&cmd, ":"));
     bool quiet = false;
-    if (!strncmp(tmp, "quiet:", 6)) {
-        cmd += 6;
+    if (cmd && !strncmp(tmp1, "quiet", 5)) {
         quiet = true;
     }
+    // Restore cmd if there was no option field
+    if (!cmd) {
+        cmd = tmp;
+    } else {
+        if (!quiet) {
+            // If there was an extra field, but with no recognizable option, assume it's a stray colon from the actual command
+            cmd = tmp;
```

Comment:
Are you sure this works? Earlier, `*cmd` was set to `tmp`, so tmp would have been modified by strsep, making this only take into account the stuff before the `:`.

Reply:
strsep modifies the content, not the pointer ;).

Reply:
Yes, exactly, which is why this would be a bug.

Reply:
> Since we never touch the original tmp pointer itself, it still points to the same address, so that works out just fine for this & free.

No, as in, the `:` is replaced with a null, cutting off the rest of the original command.

Reply:
Err, okay, this works for `free`, but you're absolutely right that this might point to data where separator may have been replaced with NULs.

(i.e., it was safe in the first branch, because no separators found means no random new NULs, but not the second one).

Reply:
And, it indeed makes much more sense to re-assign to arg directly anyway ;).

Reply:
Yep.